packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx

```diff
@@ -700,18 +707,24 @@ export class VerticalBarChartBase extends React.Component<IVerticalBarChartProps
         legendColor: this.state.color,
         shouldHighlight,
       });
-      const barHeight: number = Math.max(yBarScale(point.y), 0);
-      let adjustedBarHeight = 0;
-
-      if (barHeight <= 0) {
+      const barHeight: number = this.props.supportNegativeYValues
+        ? yBarScale(point.y) - yBarScale(0)
+        : yBarScale(point.y);
+      // Calculate threshold for minimum visible bar height
+      const minBarHeight = Math.ceil(yBarScale(this._yMax - this._yMin) / 100.0);
+      let adjustedBarHeight = barHeight;
+
+      if (Math.abs(barHeight) === 0 || (barHeight < 0 && !this.props.supportNegativeYValues)) {
         return <React.Fragment key={point.x as string}> </React.Fragment>;
-      } else if (barHeight <= Math.ceil(yBarScale(this._yMax) / 100.0)) {
-        adjustedBarHeight = Math.ceil(yBarScale(this._yMax) / 100.0);
-      } else {
-        adjustedBarHeight = barHeight;
       }
+      // Adjust bar height if it's smaller than the threshold
+      else if (Math.abs(barHeight) <= minBarHeight) {
+        adjustedBarHeight = minBarHeight * Math.sign(barHeight);
+      }
+
       const xPoint = xBarScale(point.x as number) - this._barWidth / 2;
-      const yPoint = containerHeight - this.margins.bottom! - adjustedBarHeight;
+      const yPoint = containerHeight - this.margins.bottom! - adjustedBarHeight - yBarScale(0);
+      const baselineHeight = containerHeight - this.margins.bottom! - yBarScale(0);
```

Comment:
same logic applies in these lines also

Reply:
means? ybarScale(0) needs to be replaced with ybarScale(yMax) when yMax<0, right?

Reply:
yes

Reply:
Not the internal yMax but the user defined yMax prop.